client/src/store/user/user.reducer.ts

```diff
@@ -18,25 +18,53 @@ export const userReducer = (
     case UserActionTypes.UserSignInSuccess:
       return {
         ...state,
-        user: { ...action.payload }
+        status: Status.Success
       };
 
     case UserActionTypes.UserSignInError:
       return {
         ...state,
-        user: { ...action.payload }
+        status: Status.Error,
+        errorText: action.payload.error.message
       };
 
     case UserActionTypes.UserSignUp:
       return {
         ...state,
+        status: Status.Init
       };
 
     case UserActionTypes.UserSignUpSuccess:
       return {
         ...state,
+        status: Status.Success
+       
+      };
+    case UserActionTypes.UserSignUpError:
+      return {
+        ...state,
+        status: Status.Error,
+        errorText: action.payload.error.message
+      };
+    case UserActionTypes.UserLogOut:
+      return {
+        ...state,
+        status: Status.Init,
```

Comment:
Use here 'initialUserState' from 'client/src/store/user/user.initial.ts', to not duplicate code.